dlrs/main/classes/ManageLookupRollupSummariesNewController.cls

```diff
@@ -0,0 +1,364 @@
+                parentObjFields = new List<SelectOption>();
+                ChildObjectsDuplicates = new Map<String,String>();
+                isParentObjectSelected = true;
+                childObjectList.add(new SelectOption('','-- Select --'));
+                parentObjFields.add(new SelectOption('','-- Select --')); 
+                Schema.SObjectType parentObject = objectDescription.get(selectedParentObject);
+                Schema.DescribeSObjectResult parentObjectDescribe = parentObject.getDescribe();
+                //AJB 18/11/21: changed to use a map to avoid string splitting and comparison later on
+                PnameWithRField = new Map<String, List<String>>();
+                for (Schema.ChildRelationship cr: parentObjectDescribe.getChildRelationships()) 
+                {
+                    Schema.DescribeSObjectResult dr = (cr.getChildSObject()).getDescribe(SObjectDescribeOptions.DEFERRED);
+                    String childObjectName = dr.getName();
+                    String relFieldName = String.valueOf(cr.getField());
+
+                    // AJB 16/11/21 it's actually slightly quicker to check if the map contains the key first rather than just overwriting
+                    if (!ChildObjectsDuplicates.containsKey(childObjectName) && !String.isBlank(dr.getKeyPrefix())) {
+                        ChildObjectsDuplicates.put(childObjectName, String.valueOf(dr.getLabel())+' ('+ childObjectName +')');
+                    }
+
+                    List<String> childRelationshipFields;
+                    if(!PnameWithRField.containsKey(childObjectName)) {
+                        childRelationshipFields = new List<String>();
+                    }
+                    else {
+                        childRelationshipFields = PnameWithRField.get(childObjectName);
+                        if (childRelationshipFields == null) {
+                            childRelationshipFields = new List<String>();
+                        }
+                    }
+                    childRelationshipFields.add(relFieldName);
+                    PnameWithRField.put(childObjectName, childRelationshipFields);
+                    //objectWithRelationshipField.put(childObjectName + '.'+ relFieldName, relFieldName);
+                }
+                for(String key : ChildObjectsDuplicates.keyset()){
+                    childObjectList.add(new SelectOption(key, ChildObjectsDuplicates.get(key)));
+                }
+                Map<String, schema.SObjectField> MapOfFields = parentObjectDescribe.fields.getMap();
+                for(Schema.SObjectField field : MapOfFields.values())
+                {
+                    Schema.DescribeFieldResult fr = field.getDescribe();
+                    //String dType = String.valueof(fr.getType());
+                    //if(dType == 'DOUBLE' || dType == 'INTEGER' || dType == 'CURRENCY' || dType == 'DATE' || dType == 'DATETIME' || dType == 'STRING'){
```

Comment:
totally confusing, I am afraid. 

So initially only a basic list of fields was supported. now we seem to allow all field displaytypes - are we positive this works with time fields, geo location and what not?